mysql_protocol.cc

```diff
@@ -240,21 +277,43 @@ bool Protocol::Authenticate() {
                 (const char*) bn_slave_pwhash_buf,
                 SHA_DIGEST_LENGTH)) {
       free(bn_slave_pwhash_buf);
-      SendERR(1045, "28000", "Access denied");
       return false;
     }
     free(bn_slave_pwhash_buf);
-  } else {
-    LOG(WARNING) << "No ripple_server_password_hash set, using noop authentication";
-  }
 
-  if (SendOK()) {
-    connection_->Reset();
-    connection_->SetCompressed(compress);
     return true;
+}
+
+// docs: https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase_packets_protocol_auth_switch_request.html
+bool Protocol::SendAuthSwitchRequest(const char *plugin, unsigned char *data,
+                                     size_t data_len) {
+  uint8_t packet[1 + strlen(plugin) + 1 + data_len + 1];
```

Comment:
I've tested this patch internally and it turns out we compile with -Wvla flag, i.e. variable length arrays are not allowed. Can you please change this to malloc() or make "plugin" a global constant instead of passing it into this function?